Feature/code block syntax highlighting

MarkFontenot opened the pull request:
Code blocks in droplets displayed test in light font on dark background. 

This PR updated both the draft editing mode to include syntax highlighting and the reader view to include syntax highlighting using highlight.js and a plugin for TipTap.

In the TipTap editor, I added a button for a "code chunk" within generic text block.  The editor also has a dropdown box for the creator to choose the proper language.  

I also did some updates to fix an error in the browser that seemed to be an artifact of upgrading to Next 15.  Here is a screenshot of the error.   

![image](https://github.com/user-attachments/assets/84d44e13-3519-4704-84eb-e12b4cd7ef66)

MarkFontenot committed:
Added syntax highlighting of code blocks in draft mode tiptap editor including a new button in the editor

MarkFontenot committed:
fixed some errors regarding deprecated ref stuff in next 15

MarkFontenot committed:
draft mode - added a back arrow next to droplet name in sidebar

MarkFontenot committed:
Refactor DebugToggle component to use forwardRef and remove Tooltip integration. (Trying cursor fyi)

MarkFontenot committed:
Updated the non-draft version of rendering lessons.  code blocks now syntax highlighted.

MarkFontenot committed:
Formatting Updates to pass Prettier test

MarkFontenot commented:
**Syntax Highlighting in Draft View plus the Back Button**
![2024-12-23_09-54-42](https://github.com/user-attachments/assets/26b416c5-82b8-4fb3-86e0-aca487769f17)

**Syntax Highlighting in Reader View**
![2024-12-23_09-57-01](https://github.com/user-attachments/assets/ab65a12d-cd9e-48d1-962b-ffbc5dfc41ad)

bpalazzi512 commented:
really like how it looks. Could you expand it so that you can add the blocks to expandable and callout blocks as well. Also, merge with what's in develop, as the way we do tiptap and the toolbar has changed significantly (it's a bit more organized now). 

Overall:

1. merge your changes with develop
2. add functionality to callout and expandable blocks as well

MarkFontenot committed:
Merge branch 'develop' into feature/code-block-syntax-highlighting

MarkFontenot committed:
Added code block handler and added a button to the general toolbar

MarkFontenot committed:
Added code block to Callout Block and Expandable Block.  modified utils/types/actions to handle JSON vs HTML for callout block

MarkFontenot committed:
Updated formatting for Prettier compliance

MarkFontenot commented:
- merged in changes from develop branch
- updated code base to modularize the code block with syntax highlighting to be inline with newly re-organized tiptap editor from develop
- Added code block button to general toolbar and updated functionality of expandable block and callout block to support syntax highlighted code blocks. 
- Also fixed a but with the Underline functionality in Expandable block because Underline was missing from the editor configs for expandable blocks. 

@bpalazzi512 

![2024-12-24_10-52-50](https://github.com/user-attachments/assets/009b6aba-5c36-4679-9a76-49505be75448)
![2024-12-24_10-53-10](https://github.com/user-attachments/assets/dc4678a2-67f1-48d1-9e42-b21c5dd4fe00)

bpalazzi512 committed:
Merge branch 'develop' into feature/code-block-syntax-highlighting

bpalazzi512 committed:
fixing build and tab issues

bpalazzi512 committed:
checking build

bpalazzi512 committed:
merging with develop

bpalazzi512 committed:
fixing prettier

bpalazzi512 committed:
fixing build and prettier issues
